src/irmin-pack/unix/ext.ml

```diff
@@ -250,12 +250,12 @@ module Maker (Config : Conf.S) = struct
                           (Irmin.Type.to_string XKey.t commit_key))
                   | Some (k, _kind) -> Ok k)
             in
-            let offset =
+            let end_offset, oldest_live_commit_offset =
```

Comment:
Not for this PR but I lean towards thinking this logic should live in `Gc.v` since it can be calculated from the `commit_key` that we are already passing.

Reply:
I agree, I actually did this change with your previous comments in mind :) (https://github.com/mirage/irmin/pull/2102#discussion_r987179614) we need the same logic for both `end_offset` and `oldest_live_commit_offset`, both of which can be computed from `commit_key`. I would say either both computations are done in `Gc.v` or none. wdyt?

Reply:
Ha, yeah I've been looking at this code :monocle_face: since that comment. In my mind, it would be great to move as much GC-specific code calculations as we can out of ext and into gc (even possibly the code preceding this that checks the commit key validity and the check to the file manager). So, I'd say both in `Gc.v`.

Reply:
moved `end_offset` and `oldest_live_commit_offset`, but not check of the commit key. There are some other checks done in `start`, so I think it makes sense to do all those checks in one place. Feel free to change it in subsequent PRs though.